libraries/app/include/scorum/app/database_api.hpp

```diff
@@ -196,6 +196,11 @@ class database_api
 
     std::set<std::string> lookup_budget_owners(const std::string& lower_bound_name, uint32_t limit) const;
 
+    std::vector<atomicswap_contract_api_obj> get_atomicswap_contracts(const std::string& owner) const;
+
+    atomicswap_contract_info_api_obj
+    get_atomicswap_contract(const std::string& from, const std::string& to, const std::string& secret_hash) const;
```

Comment:
why we do not get atomicswap_contract by 'id'. what is the reason asking from,to,secret_hash ?

Reply:
Due to AT protocol.